test_rclcpp/test/test_multithreaded.cpp

```diff
@@ -202,10 +203,11 @@ TEST(CLASSNAME(test_multithreaded, RMW_IMPLEMENTATION), multi_consumer_clients)
       }
       results.push_back(pair.first->async_send_request(pair.second));
     }
-    // Wait on the future produced by the first request
-    auto result = executor.spin_until_future_complete(results.back());
-
-    ASSERT_EQ(result, rclcpp::executor::FutureReturnCode::SUCCESS);
+    // Wait on each future
+    for (uint32_t i = 0; i < results.size(); i++) {
```

Comment:
I would assume that not using `size_t` for the type would result in a compiler warning on some platforms.